eudemo/eudemo/reactor.py

```diff
@@ -0,0 +1,176 @@
+# bluemira is an integrated inter-disciplinary design tool for future fusion
+# reactors. It incorporates several modules, some of which rely on other
+# codes, to carry out a range of typical conceptual fusion reactor design
+# activities.
+#
+# Copyright (C) 2022 M. Coleman, J. Cook, F. Franza, I.A. Maione, S. McIntosh, J. Morris,
+#                    D. Short
+#
+# bluemira is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# bluemira is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with bluemira; if not, see <https://www.gnu.org/licenses/>.
+
+"""
+The EUDEMO reactor design routine.
+
+1. Radial build (using PROCESS)
+2. Perform equilibria optimisation
+3. Build plasma
+4. Design scaffold for the IVCs
+5. Build vacuum vessel
+6. Build TF coils
+7. Build PF coils
+8. Build cryo thermal shield
+9. Build cryostat
+10. Build radiation shield
+11. Produce power cycle report
+"""
+
+import json
+import os
+from pathlib import Path
+from typing import Dict
+
+from bluemira.base.designer import run_designer
+from bluemira.base.parameter_frame import make_parameter_frame
+from bluemira.base.reactor import Reactor
+from bluemira.builders.divertor import Divertor, DivertorBuilder
+from bluemira.builders.plasma import Plasma, PlasmaBuilder
+from bluemira.builders.thermal_shield import VacuumVesselThermalShield, VVTSBuilder
+from bluemira.equilibria.equilibrium import Equilibrium
+from bluemira.geometry.tools import make_polygon
+from eudemo.blanket import Blanket, BlanketBuilder
+from eudemo.equilibria import EquilibriumDesigner
+from eudemo.ivc import design_ivc
+from eudemo.params import EUDEMOReactorParams
+from eudemo.pf_coils import PFCoilsDesigner
+from eudemo.radial_build import radial_build
+from eudemo.tf_coils import TFCoil, TFCoilBuilder, TFCoilDesigner
+from eudemo.vacuum_vessel import VacuumVessel, VacuumVesselBuilder
+
+CONFIG_DIR = Path(__file__).parent.parent / "config"
+PARAMS_FILE = os.path.join(CONFIG_DIR, "params.json")
+
+
+class EUDEMO(Reactor):
+    """EUDEMO reactor definition."""
+
+    plasma: Plasma
+    vacuum_vessel: VacuumVessel
+    divertor: Divertor
+    blanket: Blanket
+    tf_coils: TFCoil
+    vv_thermal: VacuumVesselThermalShield
+
```

Comment:
Some of these component managers need to be recreated. I think just `Divertor` and ` VacuumVesselThermalShield`